ros/src/common/libs/state_machine_lib/src/state_context.cpp

```diff
@@ -23,358 +23,335 @@
-{
-  if (_state)
-    return _state->getStateTransMask();
-  else
-    return 0;
-}
-
-bool StateContext::isMainState(BaseState *_state)
-{
-  return getStateKind(_state) == MAIN_STATE;
-}
-
-bool StateContext::setCurrentState(BaseState *_state)
-{
-  change_state_mutex.lock();
-  bool ret = true;
-  if (_state)
+  // create child list
+  for (auto& state : state_map_)
   {
-    uint64_t prev_state = HolderMap[getStateKind(_state)];
-    bool diff = isDifferentState(getStateNum(_state), HolderMap[getStateKind(_state)]);
-    if (isMainState(_state))
+    outputfile << state.second->getStateID() << "[label=\"" << state.second->getStateName() << "\"";
+
     {
-      if (isEmptyMainState() || enableForceSetState || (getStateTransMask(_state) & HolderMap[MAIN_STATE]))
+      std::shared_ptr<State> temp = root_state_;
+      while (temp)
       {
-        for (auto &state : HolderMap)
+        if (temp->getStateID() == state.second->getStateID())
         {
-          if (state.first == MAIN_STATE)
-          {
-            state.second = getStateNum(_state);
-          }
-          else
-          {
-            state.second = 0ULL;
-          }
+          outputfile << ",color = \"crimson\"";
         }
-      }
-      else
-      {
-        ret = false;
+        temp = temp->getChild();
       }
     }
-    else
+    if (state.second->getParent())
     {
-      if (getStateKind(_state) == BEHAVIOR_STATE)
-      {
-        HolderMap[getStateKind(_state)] |= getStateNum(_state);
-      }
-      else
-      {
-        HolderMap[getStateKind(_state)] = getStateNum(_state);
-      }
+      sublist[state.second->getParent()->getStateID()].push_back(state.second->getStateID());
     }
-    change_state_mutex.unlock();
-    if (ret && diff)
+    else
     {
-      this->inState(getStateKind(_state), prev_state);
-
-      if (getStateKind(_state) == getStateKind(prev_state) && getStateKind(_state) != BEHAVIOR_STATE)
-        if (getStateObject(prev_state))
-          getStateObject(prev_state)->outState();
+      outputfile << ", group = 1";
+      rootlist.push_back(state.second->getStateID());
     }
+    outputfile << "];\n";
   }
-  else
+
+  for (auto& root_idx : rootlist)
   {
-    change_state_mutex.unlock();
-    ret = false;
+    int idx = root_idx;
+    createGraphTransitionList(outputfile, idx, sublist);
   }
-  return ret;
-}
 
-bool StateContext::setCurrentState(uint64_t flag)
-{
-  bool ret = this->setCurrentState(getStateObject(flag));
-  return ret;
+  outputfile << "}";
 }
 
-bool StateContext::setEnableForceSetState(bool force_flag)
+std::shared_ptr<State> StateContext::getStartState()
 {
-  enableForceSetState = force_flag;
-  return true;
+  for (auto& state : state_map_)
+  {
+    if (state.second->getStateName() == "Start")
+    {
+      return state.second;
+    }
+  }
+  return nullptr;
 }
 
-std::string StateContext::getCurrentStateName(void)
+uint64_t StateContext::getStateIDbyName(std::string _name)
```

Comment:
`getStateIDbyName` returns 0  if the _for_ loop is exited without success. 0 is the assigned ID to the first state in `createStateMap`. I suggest that either `getStateIDbyName` returns something different or `createStateMap` starts assigning IDs from 1